# HG changeset patch
# User StrikerTheHedgefox <mossj32@gmail.com>
# Date 1744957272 10800
#      Fri Apr 18 03:21:12 2025 -0300
# Node ID e70a0b45601d0466be2a456b599e82d5c89ee74d
# Parent  128627054d8b04a42164c2d1c4810ce51048e4a4
# EXP-Topic strikers_stuff
Make A_ChangeFlag work on clientsided actors.

diff -r 128627054d8b -r e70a0b45601d docs/zandronum-history.txt
--- a/docs/zandronum-history.txt	Thu Apr 17 03:42:06 2025 -0300
+++ b/docs/zandronum-history.txt	Fri Apr 18 03:21:12 2025 -0300
@@ -28,6 +28,7 @@
 +	- Added "hidelives = true/false" token to SBARINFO, allowing to disable the built-in lives display on the HUD. [StrikerTheHedgefox]
 +	- Backported A_WeaponOffset from ZDoom Revision 115dbd0, with alterations necessary for Zandronum's current codebase. [StrikerTheHedgefox]
 +	- Make DISCONNECT scripts set the activator to the player's pawn, like ZDoom does. [StrikerTheHedgefox]
+-	- Fix A_ChangeFlag not working on clientsided actors. [StrikerTheHedgefox]
 -	- Fix a crash that can occur when using the "colors" DECALDEF token, when the decal's image was in the "graphics" folder. [StrikerTheHedgefox]
 -	- Fix truncation of translation index in decals. [StrikerTheHedgefox]
 
diff -r 128627054d8b -r e70a0b45601d src/thingdef/thingdef_codeptr.cpp
--- a/src/thingdef/thingdef_codeptr.cpp	Thu Apr 17 03:42:06 2025 -0300
+++ b/src/thingdef/thingdef_codeptr.cpp	Fri Apr 18 03:21:12 2025 -0300
@@ -4631,7 +4631,7 @@
 		else
 		{
 			// [BB] The server handles the flag change.
-			if ( NETWORK_InClientMode() )
+			if ( NETWORK_InClientModeAndActorNotClientHandled( self ) )
 				return;
 
 			DWORD *flagp = (DWORD*) (((char*)self) + fd->structoffset);
